hooks/pre/05-check_scenario_requirements.rb

```diff
@@ -0,0 +1,84 @@
+require 'json'
+require 'pathname'
+require 'kafo/puppet_command'
+
+PREFIX = %W(PiB TiB GiB MiB KiB B).freeze
+
+# From https://codereview.stackexchange.com/questions/9107/printing-human-readable-number-of-bytes
+def as_size(s)
+  s = s.to_f
+  i = PREFIX.length - 1
+  while s > 512 && i > 0
+    i -= 1
+    s /= 1024
+  end
+  ((s > 9 || s.modulo(1) < 0.1 ? '%d' : '%.1f') % s) + ' ' + PREFIX[i]
+end
+
+# Based on https://stackoverflow.com/a/26878510
+def is_descendant?(a, b)
+  return true if a == '/'
+
+  a_list = File.expand_path(a).split('/')
+  b_list = File.expand_path(b).split('/')
+
+  b_list[0..a_list.size-1] == a_list
+end
+
+def facts(key)
+  command = Kafo::PuppetCommand.search_puppet_path('facter')
+  raise Exception, 'Facter not found' unless command
+  JSON.load(`#{command} --json error #{key}`)[key]
```

Comment:
I also see that I left in "error" from ```--log-level error``` which was unavailable for facter 2. I'm still looking into this part but I hope that facter optimizes if you request a specific fact rather than the full data making it faster.